docs/articles/sdk/using-the-sdk-to-crosscompile.rst

```diff
@@ -33,3 +33,16 @@ Cross Compiling
 In the easiest case, when all the projects dependencies are already installed in the SDK,
 crosscompiling is really easy. Navigate in to the directory and use cmake and make to compile it as
 you normally would. The environment will make sure that the right compiler and cmake will be used.
+
+In the case of additional external dependencies, it's recommended to install them into the SDK so
+that they can be found by tools like ``pkg-config``. For example, when building a shared library
+using ``cmake``, one would typically issue these commands:
+
+.. code-block:: bash
+
+    mkdir build && cd build
+    cmake -DCMAKE_INSTALL_PREFIX="${SDKTARGETSYSROOT}/usr" ..
```

Comment:
Are you sure the CMAKE_INSTALL_PREFIX here is necessary? I don't think so, I think this is the default wher you have sourced the SDK environment. Check what `alias cmake` tells you, it should have changed and point to some tool-file which does set up a coupple of paths for cmake.

 If not then I think it shuold be.

Reply:
I've installed the SDK on my machine, and:
```
$ alias cmake
alias cmake='cmake -DCMAKE_TOOLCHAIN_FILE=/home/mardy/src/tmp/pelux/sysroots/x86_64-pokysdk-linux/usr/share/cmake/OEToolchainConfig.cmake'
```

But when I build the `template-library` project, it attempts to install it in `/usr/local`. In `CMakeCache.txt`, there's this line:
```
CMAKE_INSTALL_PREFIX:PATH=/usr/local
```
So, one really needs to override it from the command line.

Reply:
I see, yes. So I have an idea on how to make it better. What do you think about us making `CMAKE_INSTALL_PREFIX="${SDKTARGETSYSROOT}/usr` a default?

It's really easy to do because `OEToolchainConfig.cmake` is already prepared to include config subscripts. We would need to make a script in `sysroots/x86_64-pokysdk-linux/usr/share/cmake/OEToolchainConfig.cmake.d/InstallPrefix.cmake` which would set the `CMAKE_INSTALL_PREFIX="${SDKTARGETSYSROOT}/usr`. Now every time one would run the cmake alias it would use the new CMAKE_INSTALL_PREFIX instead of the `CMAKE_INSTALL_PREFIX:PATH=/usr/local`. But only if you sourced the SDK.

What do you think, does it make sense to hove the default install path to be inside the SDK if you sourced the SDK? People will still be able to overwrite it as usual.

Reply:
I'm not exactly sure that CMAKE_INSTALL is the right one, perhaps it should be CMAKE_STAGING_PREFIX instead: https://cmake.org/cmake/help/v3.0/variable/CMAKE_STAGING_PREFIX.html#variable:CMAKE_STAGING_PREFIX

Reply:
Thanks @rpannek, that sounds like a good plan! I'll play with it a bit, just to verify that it works as expected.
Which project should host the cmake override file, meta-pelux or meta-bistro?

Reply:
Not sure I understand exactly what it is you're trying to do, but most of these settings should go in a cmake toolchain file, right? We should definitely ship such a file if needed, and I suppose we can edit it when building with yocto.

Reply:
Yes, we are talking about a new file which will be automatically included by the already existing toolchain file see: http://git.yoctoproject.org/cgit/cgit.cgi/poky/tree/meta/recipes-devtools/cmake/cmake/OEToolchainConfig.cmake?h=pyro#n18

Reply:
Now with the changes in https://github.com/Pelagicore/meta-pelux/pull/50 will you change this content accordingly? Perhaps have a note that once you source the SDK you can just use make to install everything into the SDK root and why.